WooCommerce/Classes/Tools/MapsHelper.swift

```diff
@@ -0,0 +1,40 @@
+import MapKit
+
+final class MapsHelper {
+
+    /// The method accept a string of the address info you already have, and open Apple Maps on that address if found.
+    ///
+    static func openAppleMaps(address: String?, completion: @escaping (Result<Void, MapsHelperError>) -> Void) {
+        guard let address = address else {
+            completion(.failure(.locationNotFound))
+            return
+        }
+        CLGeocoder().geocodeAddressString(address) { (placemarksOptional, error) -> Void in
+              if let placemarks = placemarksOptional {
```

Comment:
This is a personal preference, but I tend to find nested if/elses difficult to read. Maybe separating the first if as a `guard let placemarks = placemarksOptional else { completion(.failure(.locationNotFound)) }` could help remove one else later on, kind of flatten things one level?